chainer/link_hooks/spectral_normalization.py

```diff
@@ -240,17 +237,22 @@ def _prepare_parameters(self, link, input_variable=None):
             # Initialize the scaling parameter with the max singular value.
             weight_matrix = self.reshape_W(initialW.array)
             # TODO(crcrpar): Remove this when chainerx supports SVD.
-            if link.xp is chainerx:
-                xp, device, array = fallback._from_chx(weight_matrix)
-                if xp is numpy:
-                    _, s, _ = numpy.linalg.svd(array)
-                else:
-                    with chainer.using_device(device):
-                        _, s, _ = xp.linalg.svd(array)
+            device = link.device
+            if device.xp is chainerx:
+                fallback_device = device.fallback_device
+                weight_matrix_ = fallback_device.send(weight_matrix)
+                with chainer.using_device(fallback_device):
+                    _, s_, _ = fallback_device.xp.linalg.svd(weight_matrix_)
+                # TODO(niboshi): s_ should be sent to `device` before being
```

Comment:
How about mentioning https://github.com/chainer/chainer/pull/7531?

Reply:
Thank you for pointing out!
I'm going to wait for #7455 and rebase after that.
This workaround will probably be unnecessary.